src/main/java/de/tum/in/www1/artemis/service/exam/ExamDeletionService.java

```diff
@@ -240,4 +256,37 @@ public void deleteTestRun(Long testRunId) {
         log.info("Request to delete Test Run {}", testRunId);
         studentExamRepository.deleteById(testRunId);
     }
+
+    /**
+     * Get the exam deletion summary for the given exam.
+     *
+     * @param examId the ID of the exam for which the deletion summary should be fetched
+     * @return the exam deletion summary
+     */
+    public ExamDeletionSummaryDTO getExamDeletionSummary(@NotNull long examId) {
+        Exam exam = examRepository.findOneWithEagerExercisesGroupsAndStudentExams(examId);
+        long numberOfBuilds = 0;
+        for (ExerciseGroup exerciseGroup : exam.getExerciseGroups()) {
+            var programmingExercises = exerciseGroup.getExercises().stream().filter(exercise -> ExerciseType.PROGRAMMING.equals(exercise.getExerciseType())).toList();
+
+            for (Exercise exercise : programmingExercises) {
+                numberOfBuilds += buildJobRepository.countBuildJobsByExerciseId(exercise.getId());
+            }
+        }
+        Channel channel = channelRepository.findChannelByExamId(examId);
+        Long conversationId = channel.getId();
+
+        List<Post> posts = postRepository.findAllByConversationId(conversationId);
+        long numberOfCommunicationPosts = posts.size();
+        long numberOfAnswerPosts = answerPostRepository.countAnswerPostsByPostIdIn(posts.stream().map(Post::getId).toList());
+
+        List<StudentExam> studentExams = studentExamRepository.findAllByExamId(examId);
```

Comment:
The student exams should be already present after calling `findOneWithEagerExercisesGroupsAndStudentExams`.